rosbag2_transport/test/rosbag2_transport/test_play_publish_clock.cpp

```diff
@@ -125,3 +127,9 @@ TEST_F(ClockPublishFixture, clock_respects_playback_rate)
   messages_to_play_ = 5;
   run_test();
 }
+
+TEST_F(ClockPublishFixture, clock_is_published_from_topic_trigger)
+{
+  play_options_.clock_publish_on_topic_publish = true;
+  run_test();
```

Comment:
Sorry my bad. I forgot that `play_options_.clock_publish_frequency`  will be initialized to 0 by default in test fixture in each test run.  
However in this case it will make sense to setup `messages_to_play_ = expected_clock_messages_;` and add a strict check inside `run_test()` for this scenario i.e. when `play_options_.clock_publish_on_topic_publish = true;`

Reply:
@kylemarcey I actually noticed that the tests from `ClockPublishFixture` test suite is tend to be flaky since `wait_for_matched` is not using there.  Also we don't need to run player in single threaded executor.   
Wouldn't you mind to rewrite `void run_test()` ?
You can refer to the right approach in   
https://github.com/ros2/rosbag2/blob/67bd7767b502eff7bac4162ff1d93e86405ff538/rosbag2_transport/test/rosbag2_transport/test_burst.cpp#L72-L87

In our case don't need to call `player->pause()` before starting playback and don't need to call `player->wait_for_playback_to_start();`